### PR TITLE
[tracing] add simplified export pipeline setup for Jaeger

### LINKED ISSUE #424: Add simplified constructor for simple trace SDK setup: Jaeger exporter
Like #343, #344, #345 but for the trace SDK. This one should cover Jaeger.

This means new methods for simplified setup, should follow the style of the metrics APIs.  NewRawExporter(), NewExportPipeline(), ...

### PR DESCRIPTION
**Pull request by matej-g**
This PR introduces a simplified exporter for Jaeger, similar to recently introduced export pipeline setups (#395) for metrics.

Resolves #424.

### CODE REVIEW

**Comment by matej-g** [comment]
> Is it possible to have tests on InstallNewPipeline/NewExportPipeline?

Sure thing, should not be a big deal, I'll provide tests as well

File: example/jaeger/main.go [hunk 1]
```diff
@@ -45,18 +45,15 @@ func initTracer() func() {
...
-		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
-		sdktrace.WithSyncer(exporter))
-	if err != nil {
-		log.Fatal(err)
-	}
-	global.SetTraceProvider(tp)
+	// For demoing purposes, use trace.AlwaysSample to sample every trace.
+	// In a production application, you should configure this to a trace.ProbabilitySampler
+	// set at the desired probability.
+	tp.ApplyConfig(sdktrace.Config{
```
Comment (jmacd) [inline example/jaeger/main.go]: I would prefer to see the `sdktrace.Config` argument passed directly to `InstallNewPipeline`. Any reason not to?

Comment (matej-g) [reply example/jaeger/main.go]: Well, I guess it depends on what is expected of `NewExportPipeline` / `InstallNewPipeline` - as this is supposed to be a simplified setup and there are default values which will be used if no config is passed to the trace provider, it is capable of returning pipeline with default configuration without needing this extra parameter.

On the other hand, I guess _default_ does not necessarily correspond to _recommended_ (and universally recommended config might not exist in this case), which is probably what the simplified pipeline setup should actually do for the user, so providing config would be reasonable as well. I was considering both implementations, so I'm not opposed.

Comment (lizthegrey) [reply example/jaeger/main.go]: we're already passing in other Jaeger properties to InstallNewPipeline, just not sdktrace.Config... so I'm on the fence here.

Comment (paivagustavo) [reply example/jaeger/main.go]: Mixing functional options with a option struct seems a little weird. 

I would love to see an unified functional options for exporters that could configure the underline SDK as well but I don't know if it is feasible without some adaptations. 

Something along the lines should be possible and I think it is better than passing the Options struct directly:
```go
// Create and install Jaeger export pipeline
tp, flush, err := jaeger.InstallNewPipeline(
	jaeger.WithCollectorEndpoint("http://localhost:14268/api/traces"),
	jaeger.WithProcess(jaeger.Process{
		ServiceName: "trace-demo",
		Tags: []core.KeyValue{
			key.String("exporter", "jaeger"),
			key.Float64("float", 312.23),
		},
	}),
	jaeger.WithSDK(
		sdktrace.AlwaysSample(),
		sdktrace.WithDefaultLabelEncoder()
	),
)
```

**Comment by rghetia** [comment]
@matej-g would you have time to address the comments?

**Comment by rghetia** [comment]
@jmacd , @paivagustavo PTAL, I have updated using consistent With* option as per the review comments.

File: exporters/trace/jaeger/jaeger.go [hunk 2]
```diff
@@ -43,6 +45,12 @@ type options struct {
 
 	//BufferMaxCount defines the total number of traces that can be buffered in memory
 	BufferMaxCount int
+
+	Config *sdktrace.Config
+
+	// Registration is set to true if the trace provider of the new pipeline should be
+	// registered as Global Trace Provider
+	Registeration bool
```
[the remark this answers is not in the document]
Comment (rghetia) [reply exporters/trace/jaeger/jaeger.go]: Either one seems fine. I will pick RegisterGlobal.

I do agree that exporter initialization should be standardized.

File: exporters/trace/jaeger/jaeger.go [hunk 3]
```diff
@@ -68,9 +76,24 @@ func WithBufferMaxCount(bufferMaxCount int) func(o *options) {
...
+// WithSDK sets the SDK config for the exporter pipeline.
+func WithSDK(config *sdktrace.Config) func(o *options) {
+	return func(o *options) {
+		o.Config = config
+	}
+}
+
+// WithRegistration enables the registration of the trace provider of the new pipeline
+// as Global Trace Provider.
+func WithRegistration() func(o *options) {
```
Comment (MrAlias) [inline exporters/trace/jaeger/jaeger.go]: Similar to changing the option name, we should change this to `WithRegisterGlobal`, right?

Or, maybe `RegisterAsGlobal`?

Comment (rghetia) [reply exporters/trace/jaeger/jaeger.go]: RegisterAsGlobal() seems better.

**Pull request merged**